tests/validation/tests/v3_api/test_vmwarevsphere_driver.py

```diff
@@ -0,0 +1,217 @@
+RANCHER_CLEANUP_CLUSTER = \
+    ast.literal_eval(os.environ.get('RANCHER_CLEANUP_CLUSTER', "True"))
+CLUSTER_NAME = os.environ.get("RANCHER_CLUSTER_NAME",
+                              random_name() + "-cluster")
+ENGINE_INSTALL_URL = os.environ.get("RANCHER_ENGINE_INSTALL_URL",
+                                    "https://get.docker.com/")
+CLONE_FROM = \
+    os.environ.get("RANCHER_CLONE_FROM",
+                   "/RNCH-HE-FMT/vm/ubuntu-bionic-18.04-cloudimg")
+RESOURCE_POOL = \
+    os.environ.get("RANCHER_RESOURCE_POOL",
+                   "/RNCH-HE-FMT/host/FMT2.R620.1/Resources/validation-tests")
+ADMIN_TOKEN = get_admin_token()
+
+rke_config = {
+    "addonJobTimeout": 30,
+    "authentication":
+        {"strategy": "x509",
+         "type": "authnConfig"},
+    "ignoreDockerVersion": True,
+    "ingress":
+        {"provider": "nginx",
+         "type": "ingressConfig"},
+    "monitoring":
+        {"provider": "metrics-server",
+         "type": "monitoringConfig"},
+    "network":
+        {"plugin": "canal",
+         "type": "networkConfig",
+         "options": {"flannelBackendType": "vxlan"}},
+    "services": {
+        "etcd": {
+            "extraArgs":
+                {"heartbeat-interval": 500,
+                 "election-timeout": 5000},
+            "snapshot": False,
+            "backupConfig":
+                {"intervalHours": 12, "retention": 6, "type": "backupConfig"},
+            "creation": "12h",
+            "retention": "72h",
+            "type": "etcdService"},
+        "kubeApi": {
+            "alwaysPullImages": False,
+            "podSecurityPolicy": False,
+            "serviceNodePortRange": "30000-32767",
+            "type": "kubeAPIService"}},
+    "sshAgentAuth": False}
+
+vsphereConfig = {
+    "cfgparam": ["disk.enableUUID=TRUE"],
+    "cloneFrom": CLONE_FROM,
+    "cloudConfig": "#cloud-config\r\npackages:\r\n - redis-server",
+    "cloudinit": "",
+    "contentLibrary": "",
+    "cpuCount": "2",
+    "creationType": "vm",
+    "customAttribute": ["203=CustomA", "204=CustomB"],
+    "datacenter": "/RNCH-HE-FMT",
+    "datastore": "/RNCH-HE-FMT/datastore/ranch01-silo01-vm01",
+    "diskSize": "20000",
+    "folder": "/",
+    "hostsystem": "",
+    "memorySize": "6000",
+    "network": ["/RNCH-HE-FMT/network/Private Range 172.16.128.1-21"],
+    "password": "",
+    "pool": RESOURCE_POOL,
+    "sshPassword": "tcuser",
+    "sshPort": "22",
+    "sshUser": "docker",
+    "sshUserGroup": "staff",
+    "tag": [
+        "urn:vmomi:InventoryServiceTag:04ffafd0-d7de-440c-a32c-5cd98761f812:GLOBAL",
+        "urn:vmomi:InventoryServiceTag:d00f1cf2-6822-46a0-9602-679ea56efd57:GLOBAL"
+    ],
+    "type": "vmwarevsphereConfig",
+    "username": "",
+    "vappIpallocationpolicy": "",
+    "vappIpprotocol": "",
+    "vappProperty": "",
+    "vappTransport": "",
+    "vcenter": "",
+    "vcenterPort": "443",
+}
+
+
+def test_valid_environment_variables():
```

Comment:
Another option would be to something like [this](http://doc.pytest.org/en/latest/skipping.html#summary) which would stop all the tests from running anyways.